apollo-federation/src/query_plan/fetch_dependency_graph.rs

```diff
@@ -1686,13 +1688,21 @@ impl FetchDependencyGraph {
             .graph
             .neighbors_directed(node_index, petgraph::Direction::Outgoing);
         let node = self.node_weight(node_index)?;
+        println!("Node: {}", node.selection_set.selection_set);
         for child_index in node_children {
             let child = self.node_weight(child_index)?;
+            println!("Child: {}", child.selection_set.selection_set);
             if node.defer_ref == child.defer_ref {
                 children.push(child_index);
             } else {
-                let parent_defer_ref = node.defer_ref.as_ref().unwrap();
+                println!(
+                    "{} == {}",
+                    DisplayOption(node.defer_ref.as_ref()),
+                    DisplayOption(child.defer_ref.as_ref())
+                );
                 let Some(child_defer_ref) = &child.defer_ref else {
+                    // FIXME: We should not be unwrapping here. Does a `DisplayOption` make sense?
+                    let parent_defer_ref = node.defer_ref.as_ref().unwrap();
                     panic!("{} has defer_ref `{parent_defer_ref}`, so its child {} cannot have a top-level defer_ref.",
                            node.display(node_index),
                            child.display(child_index),
```

Comment:
Instead of `unwrap`-ing here (`parent_defer_ref` can be `None`), we should use `DisplayOption` instead.